### DISCUSSION TITLE
局域网ip无法访问

### DISCUSSION THREAD

**Discussion by cgofun** [GaiZhenbiao/ChuanhuChatGPT#317] [Q&A] [answered]
[root@localhost ChuanhuChatGPT]# python3 ChuanhuChatbot.py 
INFO:root:获取模板文件名列表
INFO:root:获取文件名列表，目录为templates，文件类型为['.csv', 'json']，是否为纯文本列表True
INFO:root:加载模板文件1 中文Prompts.json，模式为2（0为返回字典和下拉菜单，1为返回下拉菜单，2为返回字典）
INFO:root:Loading template...
INFO:root:**您的IP区域：中国。请立即检查代理设置，在不受支持的地区使用API可能导致账号被封禁。**
/usr/local/python3/lib/python3.8/site-packages/gradio/deprecation.py:43: UserWarning: You have unused kwarg parameters in Row, please remove them: {'scale': 1}
  warnings.warn(
/usr/local/python3/lib/python3.8/site-packages/gradio/deprecation.py:43: UserWarning: You have unused kwarg parameters in Files, please remove them: {'multiple': True}
  warnings.warn(
INFO:root:获取模板文件名列表
INFO:root:获取文件名列表，目录为templates，文件类型为['.csv', 'json']，是否为纯文本列表True
INFO:root:获取模板文件名列表
INFO:root:获取文件名列表，目录为templates，文件类型为['.csv', 'json']，是否为纯文本列表True
INFO:root:获取模板文件名列表
INFO:root:获取文件名列表，目录为templates，文件类型为['.csv', 'json']，是否为纯文本列表True
INFO:root:加载模板文件1 中文Prompts.json，模式为1（0为返回字典和下拉菜单，1为返回下拉菜单，2为返回字典）
INFO:root:Loading template...
INFO:root:获取模板文件名列表
INFO:root:获取文件名列表，目录为templates，文件类型为['.csv', 'json']，是否为纯文本列表True
INFO:root:加载模板文件1 中文Prompts.json，模式为1（0为返回字典和下拉菜单，1为返回下拉菜单，2为返回字典）
INFO:root:Loading template...
INFO:root:获取历史记录文件名列表
INFO:root:获取文件名列表，目录为history，文件类型为['.json']，是否为纯文本列表True
INFO:root:获取历史记录文件名列表
INFO:root:获取文件名列表，目录为history，文件类型为['.json']，是否为纯文本列表True
INFO:root:
川虎的温馨提示：访问 http://localhost:7860 查看界面
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
START /usr/bin/firefox "http://127.0.0.1:7860/"
Running without a11y support!

**Comment by Keldos-Li** [comment]
你是在Linux命令行中直接部署的？这个程序需要GUI支持。

**Comment by cgofun** [comment]
已经成功了，哈哈

**Reply by Keldos-Li** [reply]
可以贴一下解决方法，让后人有个参考~

**Comment by cgofun** [comment, marked as answer]
首先python版本要3.8以上，然后修改ChuanhuChatbot.py中的非docker模式两句中没有指定监听地址和端口，加入并改成自己需要的的就可以了。
        if authflag:
            demo.queue().launch(server_name="0.0.0.0", server_port=80, share=True, auth=(username, password), favicon_path="./assets/favicon.png", inbrowser=True)
        else:
            demo.queue().launch(server_name="0.0.0.0", server_port=80, share=True, favicon_path="./assets/favicon.png", inbrowser=True)  # 改为 share=True 可以创建公开分享链接

**Reply by Keldos-Li** [reply]
额，好吧，确实是要自己加的。
